src/main/java/frc/robot/subsystems/BallPath/Shooter/ShooterImpl.java

```diff
@@ -2,17 +2,83 @@
 
 import java.util.concurrent.TimeUnit;
 
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+
 // import ca.team3161.lib.robot.subsystem.RepeatingIndependentSubsystem;
 import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
+import ca.team3161.lib.utils.Utils;
+// import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
 public class ShooterImpl extends RepeatingPooledSubsystem implements Shooter {
-    public ShooterImpl() {
+    
+    private TalonSRX turretMotor;
+    private TalonFX shooterMotor;
+    private TalonSRX hoodMotor;
+
+    private double kp = 0.00035;
+    private double ki = 0.000075;
+    // private final double kd = -0.0002;  Value that was being tested as of February 3, 2022.
+    private double kd = 0.00003;
+
+    private double setPointHood;
+    private double setPointShooter;
+    private double setPointRotation;
+    private double setPointShooterPID;
+
+    private double shooterEncoderReadingPosition;
+    double shooterEncoderReadingVelocity;
+    private double turretHoodPosition;
+    private double turretEncoderReadingPosition;
+    private double turretHoodVelocity;
+    private double turretEncoderReadingVelocity;
+
+    private boolean turretReady = false;
+    private boolean hoodReady = false;
+
+    double currentOutput;
+
+    private final double hoodBuffer = 20000;
+    private final double turretBuffer = 30000;
+    private final double turretSpeed = 0.3;
+    private final double hoodSpeed = 0.3;
+
+
+    private final PIDController shooterPid = new PIDController(kp, ki, kd);
+
+    public ShooterImpl(TalonSRX turretMotor, TalonFX shooterMotor, TalonSRX hoodMotor) {
         super(10, TimeUnit.MILLISECONDS);
         // TODO Auto-generated constructor stub
+        this.turretMotor = turretMotor;
+        this.shooterMotor = shooterMotor;
+        this.hoodMotor = hoodMotor;
+
+        shooterEncoderReadingPosition = shooterMotor.getSelectedSensorPosition();
+        shooterEncoderReadingVelocity = shooterMotor.getSelectedSensorVelocity();
+        turretHoodPosition = hoodMotor.getSelectedSensorPosition();
+        turretHoodVelocity = hoodMotor.getSelectedSensorVelocity();
+        turretEncoderReadingPosition = this.turretMotor.getSelectedSensorPosition();
+        turretEncoderReadingVelocity = this.turretMotor.getSelectedSensorVelocity();
```

Comment:
All of these positions/velocities will only get read and assigned once, when the subsystem is first initialized. I think moving these assignments and the `SmartDashboard.putNumber`s into the `task()` probably makes more sense?